### ISSUE TITLE
PodSecurity: do not require `runAsNonRoot: true` if a non-zero `runAsUser` value is provided

### ISSUE THREAD

**Issue by liggitt** [kubernetes/kubernetes#105186]
#### What would you like to be added:

Relax the `restricted` Pod Security Standard to accept either of `runAsNonRoot: true` or a non-zero `runAsUser` value

#### Why is this needed:

It seems silly to require a pod to set `runAsNonRoot: true` to pass the restricted profile when it has already explicitly set `runAsUser: <non-zero-value>`. I ran into this putting together pod security benchmarks, using a kube-apiserver manifest that was already configured to run as a non-root user, and it failed the restricted PodSecurity check.

/assign
/cc @tallclair 
/sig auth security
/milestone v1.23

**Comment by liggitt** [comment]
cc @kubernetes/sig-security-pr-reviews @kubernetes/sig-auth-pr-reviews

**Comment by liggitt** [comment]
hmm... actually, it looks like the kubelet has added windows-specific enforcement around this: 

https://github.com/kubernetes/kubernetes/blob/2bfb2eba80824ad14a2b6838b3a8bc602fc7f4e6/pkg/kubelet/kuberuntime/security_context_windows.go#L34-L40

I'm not actually sure that matches the expectations of people using the field, or how the default container image username aspect maps on windows...

**Comment by sftim** [comment]
Yet Another Defaulting Admission Controller? (to infer `runAsNonRoot` from a non-zero `runAsUser` value).

&mdash;would that even help?

**Comment by raesene** [comment]
On the overall point, I'd agree needing both to be set would likely confuse users, setting a non-root UID in `runAsUser` implies the `runAsNonRoot: true` piece.

I'd guess that yep a mutating admission controller to add an `runAsNonRoot: true` where a `runAsUser` value was set to not been UID 0 would work, but it feels pretty hacky.

Also that Windows setup looks ... odd I wouldn't usually equate `containerAdministrator` as posing the same risks as `root` from a security perspective.

**Comment by liggitt** [comment]
Yeah... the windows-specific behavior is undocumented... I think I would still lean toward making the `restricted` level require `runAsNonRoot: true` or a non-zero `runAsUser`.

**Comment by tallclair** [comment]
Should we re-open the original issue (don't require `runAsNonRoot: true` if a non-zero `runAsUser` value is provided) once PodOS goes GA? Should this be handled as a line-item in https://github.com/kubernetes/kubernetes/pull/105919?

**Comment by liggitt** [comment]
you mean if the OS is explicitly set to linux?
